src/reducers/sources.js

```diff
@@ -69,14 +72,14 @@ function update(state = State(), action: Action) : Record<SourcesState> {
       });
 
     case "CLOSE_TAB":
-      availableTabs = removeSourceFromTabList(state.tabs, action.id);
+      availableTabs = removeSourceFromTabList(state, state.tabs, action.id);
```

Comment:
the biggest thing I see missing here is that I would add a new case for when the debugger starts. I'm not sure what the action is, but if you put a breakpoint in the reducer you should be able to observe one of the early actions and hook in there.

```js
case "???":
  // load the tabs
```